src/web3-utils.js

```diff
@@ -130,3 +130,20 @@ export function isValidEnsName(name) {
 
 // Re-export some utilities from web3-utils
 export { fromWei, isAddress, toChecksumAddress, toWei } from 'web3-utils'
+
+/*
+ * Return the injected provider, if any.
+ */
+export function getInjectedProvider() {
+  // Status implements EIP 1102 using `ethereumBeta`
+  if (window.ethereumBeta) {
+    return window.ethereumBeta
+  }
+  if (window.ethereum) {
+    return window.ethereum
+  }
+  if (window.web3 && window.web3.currentProvider) {
+    return window.web3.currentProvider
+  }
```

Comment:
Wondering if there's any point in using `eth-provider` if we check for `window.ethereum` and `window.web3.currentProvider` here; [`eth-provider` basically does the same check](https://github.com/floating/eth-provider/blob/master/browser.js#L15).

Didn't quite follow the mess up on the `.send()` / `.emit()` API, but are cipher and coinbase not working with eth-provider due to their injected providers not having specific methods?

Reply:
We should then just descope `eth-provider` to only look for frame and localrpc connections.

Wondering about handling EIP-1102; I believe `eth-provider` did that automatically before and we'll need to reimplement it if we don't use it?

Reply:
It looks like `eth-provider` only uses the `emit()` internally, and doesn't seem to be using those status updates at all: https://github.com/floating/eth-provider/issues/19

Reply:
@AquiGorka `eth-provider` should have the `.emit()` issue fixed now; can you try bumping to `eth-provider@0.2.0` and see if there are still problems with using it?

Reply:
@sohkai tried without this fix and with `eth-provider@0.2.0` and it went back to "not working" (trying to cast a vote I get the "unlock and enable provider" message).